compiler/frontend/beam/src/main/java/org/apache/nemo/compiler/frontend/beam/PipelineTranslator.java

```diff
@@ -64,14 +73,18 @@
   private final Map<Class<? extends PTransform>, Method> primitiveTransformToTranslator = new HashMap<>();
   private final Map<Class<? extends PTransform>, Method> compositeTransformToTranslator = new HashMap<>();
 
+  private static final AtomicReference<Pipeline> PIPELINE = new AtomicReference<>();
```

Comment:
How about moving this into `TranslationContext` to reduce its access scope, similar to `final CompositeTransformVertex pipeline`?

Reply:
A separate issue sounds good. Thanks!

Reply:
[Issue-220](https://issues.apache.org/jira/browse/NEMO-220) created